admin/server.go

```diff
@@ -0,0 +1,55 @@
+package admin
+
+import (
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+
+	"github.com/reddit/baseplate.go/log"
+)
+
+const (
+	DefaultAdminAddr = ":6060"
+)
+
+// ServerArgs contain optional configuration for the admin server.
+type ServerArgs struct {
+	// AdminAddr is a custom address for the admin server. Defaults to port 6060 on localhost if not set.
+	AdminAddr string
+	// HealthCheckFn is the HTTP handler for health check.
+	HealthCheckFn func(w http.ResponseWriter, r *http.Request)
+}
+
+// NewServer returns a new admin server for internal functionality.
+func NewServer(args *ServerArgs) *Server {
+	adminAddr := args.AdminAddr
+	if args.AdminAddr == "" {
+		adminAddr = DefaultAdminAddr
+	}
+
+	return &Server{
+		adminAddr:     adminAddr,
+		healthCheckFn: args.HealthCheckFn,
+	}
+}
+
+type Server struct {
+	adminAddr     string
+	healthCheckFn func(w http.ResponseWriter, r *http.Request)
+}
+
+// Serve starts an HTTP server for internal functions:
+//    metrics       - serve /metrics for prometheus
+//    health check  - serve /health for health checking
+//    pprof         - https://blog.golang.org/pprof
+// Default server address is http://localhost:6060.
+func (s *Server) Serve() {
+	go func() {
+		if s.healthCheckFn != nil {
+			http.HandleFunc("/health", s.healthCheckFn)
```

Comment:
Is it worth having a default for the health check fn as well?

Reply:
this is only meant for http services. for thrift services providing another http health check endpoint could gets confusing.

Reply:
I think a default handler would be useful. It would allow users to always include a service health check in their supervisor (i.e. Kubernetes pod health checks).

Even if it's an always-ok stub, it allows for consistency of deployment configuration.

Reply:
I'm fine with either adding a default health check or not. @fishy do you have a preference?

Reply:
We might want to do that eventually, but now is not the time yet (we need more discussion, etc.. there's already a discussion on how the v2 health check can be provided by the framework/library), so we should leave that out for a future PR.